server/src/main/java/org/opensearch/cluster/coordination/UnsafeBootstrapMasterCommand.java

```diff
@@ -134,9 +129,16 @@ protected void processNodePaths(Terminal terminal, Path[] dataPaths, int nodeLoc
         Settings persistentSettings = Settings.builder()
             .put(metadata.persistentSettings())
             .put(UNSAFE_BOOTSTRAP.getKey(), true)
-            .put(Metadata.SETTING_READ_ONLY_SETTING.getKey(), applyClusterReadOnlyBlock)
             .build();
 
+        Boolean applyClusterReadOnlyBlock = applyClusterReadOnlyBlockOption.value(options);
+        if(Objects.nonNull(applyClusterReadOnlyBlock)) {
```

Comment:
I am wondering if the user passes false explicitly, then what should be the behavior. Should we keep the existing block as is (in case it exists on the cluster)? Thoughts?

Reply:
I think we should entertain what user has given in input, because unsafe-bootstrap should be used consciously and whoever is running the command should know what it will do. Also if cluster block is already present then we hope that user will aware about the same while using this command.